From f1f8920d798dbcb20cb775b46a54cd81847295fd Mon Sep 17 00:00:00 2001
From: Gavin Smith <gavinsmith0123@gmail.com>
Date: Tue, 14 Nov 2023 21:53:49 +0000
Subject: [PATCH 5/5] * tp/Texinfo/command_data.txt (item_LINE, itemx,
 defblock, defline, deftypeline): Remove contain_basic_inline flag.  There is
 no reason an @anchor should not occur inside @item, inside @table, or the
 other commands, as no index entry is being created with the @anchor.

Report from Ihor Radchenko <yantar92@posteo.net> for Org mode manual.
---
 ChangeLog                                     | 10 +++
 tp/Texinfo/command_data.txt                   | 10 +--
 tp/t/results/invalid_nestings/in_table.pl     | 90 -------------------
 .../invalid_nestings/table_on_item_line.pl    |  9 --
 4 files changed, 15 insertions(+), 104 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index a146820671..0dcdb1a904 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,13 @@
+2023-11-14  Gavin Smith <gavinsmith0123@gmail.com>
+
+	* tp/Texinfo/command_data.txt
+	(item_LINE, itemx, defblock, defline, deftypeline):
+	Remove contain_basic_inline flag.  There is no reason an @anchor
+	should not occur inside @item, inside @table, or the other
+	commands, as no index entry is being created with the @anchor.
+
+	Report from Ihor Radchenko <yantar92@posteo.net> for Org mode manual.
+
 2023-11-13  Gavin Smith <gavinsmith0123@gmail.com>
 
 	* tp/Texinfo/XS/parsetexi/tree.c (reset_obstacks):
diff --git a/tp/Texinfo/command_data.txt b/tp/Texinfo/command_data.txt
index bcda173e2c..c9b5f51569 100644
--- a/tp/Texinfo/command_data.txt
+++ b/tp/Texinfo/command_data.txt
@@ -253,8 +253,8 @@ printindex              line,formattable_line,close_paragraph,global,contain_pla
 listoffloats            line,formattable_line,close_paragraph,global,contain_basic_inline   LINE_line
 exdent                  line,formatted_line,close_paragraph            LINE_line
 # or nobrace skipspace, depending on the context
-item_LINE               line,formatted_line,close_paragraph,contain_basic_inline    LINE_line
-itemx                   line,formatted_line,close_paragraph,contain_basic_inline    LINE_line
+item_LINE               line,formatted_line,close_paragraph    LINE_line
+itemx                   line,formatted_line,close_paragraph    LINE_line
 nodedescription         line,close_paragraph                               LINE_line
 # in index entries
 subentry                line,in_index,contain_basic_inline          LINE_line
@@ -494,9 +494,9 @@ defmethod               block,def,contain_basic_inline,def_alias,close_paragraph
 deftypemethod           block,def,contain_basic_inline,def_alias,close_paragraph    BLOCK_def
 
 # generic, no automatic index
-defblock                block,contain_basic_inline,close_paragraph       BLOCK_def
-defline                 line,def,contain_basic_inline,close_paragraph        LINE_line
-deftypeline             line,def,contain_basic_inline,close_paragraph        LINE_line
+defblock                block,close_paragraph           BLOCK_def
+defline                 line,def,close_paragraph        LINE_line
+deftypeline             line,def,close_paragraph        LINE_line
 
 # def*x
 deffnx                  line,def,contain_basic_inline,close_paragraph        LINE_line
diff --git a/tp/t/results/invalid_nestings/in_table.pl b/tp/t/results/invalid_nestings/in_table.pl
index f4dcef1141..76eea8b3b4 100644
--- a/tp/t/results/invalid_nestings/in_table.pl
+++ b/tp/t/results/invalid_nestings/in_table.pl
@@ -1107,42 +1107,6 @@ $result_errors{'in_table'} = [
     'text' => '@indent should not appear in @item',
     'type' => 'warning'
   },
-  {
-    'error_line' => 'warning: @indent should not appear on @item line
-',
-    'file_name' => '',
-    'line_nr' => 9,
-    'macro' => '',
-    'text' => '@indent should not appear on @item line',
-    'type' => 'warning'
-  },
-  {
-    'error_line' => 'warning: @titlefont should not appear on @item line
-',
-    'file_name' => '',
-    'line_nr' => 9,
-    'macro' => '',
-    'text' => '@titlefont should not appear on @item line',
-    'type' => 'warning'
-  },
-  {
-    'error_line' => 'warning: @anchor should not appear on @item line
-',
-    'file_name' => '',
-    'line_nr' => 9,
-    'macro' => '',
-    'text' => '@anchor should not appear on @item line',
-    'type' => 'warning'
-  },
-  {
-    'error_line' => 'warning: @footnote should not appear on @item line
-',
-    'file_name' => '',
-    'line_nr' => 9,
-    'macro' => '',
-    'text' => '@footnote should not appear on @item line',
-    'type' => 'warning'
-  },
   {
     'error_line' => 'warning: @exdent should only appear at the beginning of a line
 ',
@@ -1161,24 +1125,6 @@ $result_errors{'in_table'} = [
     'text' => '@exdent should not appear in @item',
     'type' => 'warning'
   },
-  {
-    'error_line' => 'warning: @exdent should not appear on @item line
-',
-    'file_name' => '',
-    'line_nr' => 9,
-    'macro' => '',
-    'text' => '@exdent should not appear on @item line',
-    'type' => 'warning'
-  },
-  {
-    'error_line' => 'warning: @ref should not appear on @item line
-',
-    'file_name' => '',
-    'line_nr' => 11,
-    'macro' => '',
-    'text' => '@ref should not appear on @item line',
-    'type' => 'warning'
-  },
   {
     'error_line' => '@ref missing closing brace
 ',
@@ -1206,15 +1152,6 @@ $result_errors{'in_table'} = [
     'text' => '@center should not appear in @item',
     'type' => 'warning'
   },
-  {
-    'error_line' => 'warning: @center should not appear on @item line
-',
-    'file_name' => '',
-    'line_nr' => 13,
-    'macro' => '',
-    'text' => '@center should not appear on @item line',
-    'type' => 'warning'
-  },
   {
     'error_line' => 'warning: @cindex should not appear in @item
 ',
@@ -1224,15 +1161,6 @@ $result_errors{'in_table'} = [
     'text' => '@cindex should not appear in @item',
     'type' => 'warning'
   },
-  {
-    'error_line' => 'warning: @cindex should not appear on @item line
-',
-    'file_name' => '',
-    'line_nr' => 14,
-    'macro' => '',
-    'text' => '@cindex should not appear on @item line',
-    'type' => 'warning'
-  },
   {
     'error_line' => 'warning: @cindex should not appear in @item
 ',
@@ -1242,15 +1170,6 @@ $result_errors{'in_table'} = [
     'text' => '@cindex should not appear in @item',
     'type' => 'warning'
   },
-  {
-    'error_line' => 'warning: @cindex should not appear on @item line
-',
-    'file_name' => '',
-    'line_nr' => 18,
-    'macro' => '',
-    'text' => '@cindex should not appear on @item line',
-    'type' => 'warning'
-  },
   {
     'error_line' => 'warning: @cindex should not appear in @item
 ',
@@ -1260,15 +1179,6 @@ $result_errors{'in_table'} = [
     'text' => '@cindex should not appear in @item',
     'type' => 'warning'
   },
-  {
-    'error_line' => 'warning: @cindex should not appear on @item line
-',
-    'file_name' => '',
-    'line_nr' => 21,
-    'macro' => '',
-    'text' => '@cindex should not appear on @item line',
-    'type' => 'warning'
-  },
   {
     'error_line' => 'warning: empty index key in @item
 ',
diff --git a/tp/t/results/invalid_nestings/table_on_item_line.pl b/tp/t/results/invalid_nestings/table_on_item_line.pl
index b1184ba915..51ba523fd7 100644
--- a/tp/t/results/invalid_nestings/table_on_item_line.pl
+++ b/tp/t/results/invalid_nestings/table_on_item_line.pl
@@ -273,15 +273,6 @@ $result_errors{'table_on_item_line'} = [
     'text' => '@table should not appear in @item',
     'type' => 'warning'
   },
-  {
-    'error_line' => 'warning: @table should not appear on @item line
-',
-    'file_name' => '',
-    'line_nr' => 2,
-    'macro' => '',
-    'text' => '@table should not appear on @item line',
-    'type' => 'warning'
-  },
   {
     'error_line' => 'no matching `@end table\'
 ',
-- 
2.42.1

